File: androidApp/src/test/java/com/commuto/interfacemobile/android/offer/TestOfferTruthSource.kt
package com.commuto.interfacemobile.android.offer

import androidx.compose.runtime.mutableStateMapOf
import androidx.compose.runtime.mutableStateOf
import java.math.BigInteger
import java.util.*

/**
 * A basic [OfferTruthSource] implementation for testing.
 */
class TestOfferTruthSource: OfferTruthSource {
    override var offers = mutableStateMapOf<UUID, Offer>()
    override var serviceFeeRate = mutableStateOf<BigInteger?>(null)
    override fun addOffer(offer: Offer) {
        offers[offer.id] = offer
    }
    override fun removeOffer(id: UUID) {
        offers.remove(id)
    }
}

File: iosApp/iosAppTests/offer/TestOfferTruthSource.swift
//
//  TestOfferTruthSource.swift
//  iosAppTests
//
//  Created by jimmyt on 9/3/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import BigInt
import Foundation
@testable import iosApp

/**
 A basic `OfferTruthSource` implementation for testing.
 */
class TestOfferTruthSource: OfferTruthSource {
    var serviceFeeRate: BigUInt?
    var offers: [UUID: Offer] = [:]
}
